### PR TITLE
Add support for Third Party Resources

### PR DESCRIPTION
**Pull request by jonathan-kosgei**
Hi,
Here's a basic PR that supports watching third party resources.

### CODE REVIEW

**Comment by jonathan-kosgei** [comment]
Signed CLA

File: scripts/preprocess_spec.py [hunk 1]
```diff
@@ -102,7 +103,18 @@ def strip_tags_from_operation_id(operation, _):
     operation['operationId'] = operation_id
 
 
+def add_thirdparty_resource_paths(spec):
```
Comment (mbohlool) [inline scripts/preprocess_spec.py]: can you convert the file to json and remove dependency on yaml?

**Comment by mbohlool** [comment]
Thanks for the PR.
1. Can you add an example?
2. Remove dependency to yaml. json is already there.
3. stash commits together to have 2-3 meaningful commits.

File: scripts/preprocess_spec.py (hunk 1 above)
Comment (jonathan-kosgei) [inline scripts/preprocess_spec.py]: sure, on it

Comment (jonathan-kosgei) [inline scripts/preprocess_spec.py]: done

**Comment by jonathan-kosgei** [comment]
I have an example ready, should I squash your commits as well as mine? Can I do that?

**Comment by mbohlool** [comment]
My commits should have not got to your PR. I guess you did a bad rebase or something. you need to do some git magic to only have your commits there. Do a `git fetch upstream` where upstream set to the original repo and then `git rebase upstream/master` then hopefully those non related commits will go away and then do a git rebase -i HEAD~X where X is the exact number of your commit and then squash/edit commits.

**Comment by jonathan-kosgei** [comment]
Added, put/delete and create.
I ran into a couple of merge conflicts and the commits are still there, I'm thinking of creating a new pull request and discarding this one. The changes I made were to 4 files so they won't be too much work to recreate.. what do you think?

**Comment by mbohlool** [comment]
Normally in this situation, it is easiest think to do. I can be fixed, but not worth it. Just create a new one :)

**Comment by jonathan-kosgei** [comment]
Closed in favor of new PR

**Pull request closed without merging**